src/repository/company-repository.ts

```diff
@@ -0,0 +1,51 @@
+import { Company } from "../model/company";
+import { CompanyParams } from "../util/types";
+
+export class CompanyRepository {
+  async createCompany(companyParams: CompanyParams) {
+    try {
+      return Company.create({ ...companyParams}).save();
+    } catch (err) {
+      console.error(err); 
+      throw new Error("Creating company failed!");
+    }
+  }
+
+  async deleteCompany(companyId: number) {
+    try {
+      const company = await Company.findOne({
```

Comment:
You have function "getCompanyById" that does the same thing. Is better to use that function to get the company.

Reply:
Changed it everywhere.